lib/galaxy/webapps/galaxy/api/cloudauthz.py

```diff
@@ -146,3 +146,29 @@ def create(self, trans, payload, **kwargs):
             log.exception(msg_template.format("exception while creating the new cloudauthz record"))
             raise InternalServerError('An unexpected error has occurred while responding to the create request of the '
                                       'cloudauthz API.' + str(e))
+
+    @expose_api
+    def delete(self, trans, encoded_authz_id, **kwargs):
+        """
+        * DELETE /api/cloud/authz/{encoded_authz_id}
+            Deletes the CloudAuthz record with the given ``encoded_authz_id``.
+        """
+
+        msg_template = "Rejected user `" + str(trans.user.id) + "`'s request to delete cloudauthz config because of {}."
+        try:
+            authz_id = self.decode_id(encoded_authz_id)
+        except Exception:
+            log.debug(msg_template.format("cannot decode authn_id `" + str(authz_id) + "`"))
```

Comment:
You probably mean for this to be `encoded_authz_id`?

Reply:
(and in the line below, `authz_id`, instead of `authn_id`?)

Reply:
good catch, I guess too many `authz`, `authn`, and `authnz` ;-)